granulate_utils/linux/cgroups/cgroup.py

```diff
@@ -78,7 +101,94 @@ def find_v2_hierarchy(resolve_host_root_links: bool = True) -> Optional[str]:
     return path
 
 
-def get_cgroup_mount(controller: str, resolve_host_root_links: bool = True) -> Optional[str]:
+CGROUP_PROCS_FILE = "cgroup.procs"
+
+
+class CgroupCore:
+    path: Path
+
+    def __init__(self, path: Path):
+        self.path = path
+
+    def _get_parent_cgroup(self, cgroup_name: str) -> Optional[Path]:
+        if cgroup_name in self.path.parts:
+            return Path(self.path.as_posix().split(cgroup_name)[0]) / cgroup_name
+        return None
+
+    def _get_cgroup_in_hierarchy(self, cgroup_name: str) -> Optional[Path]:
+        if self.has_parent_cgroup(cgroup_name):
+            new_cgroup_path = self._get_parent_cgroup(cgroup_name)
```

Comment:
Btw, is there any expected case where this will return an ancestor cgroup? In normal cases (unless we have bugs) it'll always be CURRENT cgroup or no cgroup exists and then we create a child, right?

Reply:
This isn't an expected case but generally it could happen. Thinking about it, it's probably not that interesting and we  should just check the current cgroup and not the entire hierarchy.

Reply:
The case I see is a bug if we re-create the same cgroup recursively.

Let's keep it simple and remove the parent logic - check just the current.